applications/nrf_desktop/doc/porting_guide.rst

```diff
@@ -1,10 +1,152 @@
+   Pay attention to pins that are used and to the bus configuration for optical sensor.
+#. Copy the project files for the device that is the closest match for your hardware.
+   For example, for gaming mouse these are located at ``applications/nrf_desktop/configure/nrf52840_pca20041``.
+#. Optionally, depending on the board, edit the DTS overlay file.
+   This step is not required if you have created a new board and its DTS files fully describe your hardware.
+   In such case, the overlay file can be left empty.
+#. In Kconfig, ensure that the following modules that are specific for gaming mouse are enabled:
+
+    * :ref:`motion`
+    * :ref:`wheel`
+    * :ref:`buttons`
+    * :ref:`battery_meas`
+    * :ref:`leds`
+
+#. For each module enabled, change its ``_def`` file to match your hardware:
+
+    * Motion module
+        * The ``nrf52840_pca20041`` uses the PMW3360 optical motion sensor.
+          The sensor is configured in DTS, and the sensor type is selected in the application configuration.
+          If you need to add a new sensor, expand the application configuration.
+    * wheel module
+        * The wheel is based on the QDEC peripheral of the nRF52840 device and the hardware-related part is configured in DTS.
+    * buttons module
+        * To simplify the configuration of arrays, the nRF Desktop project uses ``_def`` files.
+        * The ``buttons`` module's ``_def`` file contains pins assigned to rows and columns.
+    * GATT Battery module
+        * The battery module ``_def`` file contains mapping needed to match the voltage read from ADC to the battery level.
+    * leds module
+        * The application uses two LEDs -- one for the peers state, and one for the system state indication.
+        * To control each LED brightness, the project uses the PWM channels.
+          The PWM peripheral needs to be configured in DTS files, and the leds module ``_def`` file needs to be configured to indicate which PWM channel is assigned to each LED color.
+
+#. Review Bluetooth options in Kconfig.
+   Refer to the Bluetooth configuration page for the list of available options.
+   Ensure that the Bluetooth role is configured appropriately.
+   For mouse, it should be configured as peripheral.
+#. Edit Kconfig to disable options that you do not use.
+   Some options have dependencies that may no longer be needed when other options are disabled.
+   For example, when the LEDs module is disabled, PWM driver will also not be needed.
+
+
+Application modules
+*******************
+
+For more information about each application module and its configuration details, see the following pages:
 
 .. toctree::
-   :maxdepth: 1
+   :maxdepth: 2
 
    hw_interface.rst
-   modules.rst
\ No newline at end of file
+   modules.rst
+
+.. |preconfigured_build_types| replace:: The preconfigured build types configure the device with or without the bootloader and in debug or release mode.
```

Comment:
I think this should be moved to nrf/doc/nrf/shortcuts.txt

Reply:
build types are our invention as neither zephyr nor NCS actually use them (though it would be great if others use them too)
I don't know how to handle this, maybe @greg-fer can help?